Tests/SubprocessTests/ProcessMonitoringTests.swift

```diff
@@ -0,0 +1,361 @@
+        )
+        let spawnResult = try config.spawn(
+            withInput: self.devNullInputPipe(),
+            outputPipe: self.devNullOutputPipe(),
+            errorPipe: self.devNullOutputPipe()
+        )
+        defer {
+            spawnResult.execution.processIdentifier.close()
+        }
+
+        async let monitorResult = try monitorProcessTermination(
+            for: spawnResult.execution.processIdentifier
+        )
+        // Send signal to process
+        try spawnResult.execution.send(signal: .terminate)
+
+        let result = try await monitorResult
+        #expect(result == .unhandledException(SIGTERM))
+    }
+    #endif
+}
+
+// MARK: - Edge Cases
+extension SubprocessProcessMonitoringTests {
+    @Test func testAlreadyTerminatedProcess() async throws {
+        let config = self.immediateExitProcess(withExitCode: 0)
+        let spawnResult = try config.spawn(
+            withInput: self.devNullInputPipe(),
+            outputPipe: self.devNullOutputPipe(),
+            errorPipe: self.devNullOutputPipe()
+        )
+        defer {
+            spawnResult.execution.processIdentifier.close()
+        }
+        // Manually wait for the process to make sure it exits
+        #if os(Windows)
+        WaitForSingleObject(
+            spawnResult.execution.processIdentifier.processDescriptor,
+            INFINITE
+        )
+        #else
+        var siginfo = siginfo_t()
+        waitid(
+            P_PID,
+            id_t(spawnResult.execution.processIdentifier.value),
+            &siginfo,
+            WEXITED | WNOWAIT
+        )
+        #endif
+        // Now make sure monitorProcessTermination() can still get the correct result
+        let monitorResult = try await monitorProcessTermination(
+            for: spawnResult.execution.processIdentifier
+        )
+        #expect(monitorResult == .exited(0))
+    }
+
+    @Test func testCanMonitorLongRunningProcess() async throws {
+        let config = self.longRunningProcess(withTimeOutSeconds: 1)
+        let spawnResult = try config.spawn(
+            withInput: self.devNullInputPipe(),
+            outputPipe: self.devNullOutputPipe(),
+            errorPipe: self.devNullOutputPipe()
+        )
+        defer {
+            spawnResult.execution.processIdentifier.close()
+        }
+        let monitorResult = try await monitorProcessTermination(
+            for: spawnResult.execution.processIdentifier
+        )
+
+        #expect(monitorResult.isSuccess)
+    }
+
+    @Test func testInvalidProcessIdentifier() async throws {
+        #if os(Windows)
+        let expectedError = SubprocessError(
+            code: .init(.failedToMonitorProcess),
+            underlyingError: .init(rawValue: DWORD(ERROR_INVALID_PARAMETER))
+        )
+        let processIdentifier = ProcessIdentifier(
+            value: .max, processDescriptor: INVALID_HANDLE_VALUE, threadHandle: INVALID_HANDLE_VALUE
+        )
+        #elseif os(Linux)
```

Comment:
```suggestion
        #elseif os(Linux) || os (FreeBSD)
```

I have the code to restore FreeBSD support on a branch; one less thing to update.